src/Stratis.Bitcoin/P2P/PeerAddressManagerBehaviour.cs

```diff
@@ -17,6 +17,24 @@ namespace Stratis.Bitcoin.P2P
     /// </summary>
     public sealed class PeerAddressManagerBehaviour : NetworkPeerBehavior
     {
+        /// <summary>Provider of time functions.</summary>
+        private readonly IDateTimeProvider dateTimeProvider;
+
+        /// <summary>
+        /// See <see cref="PeerAddressManagerBehaviourMode"/> for the different modes and their
+        /// explanations.
+        /// </summary>
+        public PeerAddressManagerBehaviourMode Mode { get; set; }
+
+        /// <summary>Peer address manager instance, see <see cref="IPeerAddressManager"/>.</summary>
```

Comment:
you don't really need to tell the reader to see the type of the thing you are commenting. see should be used when you are talking about something else, ... same above with PeerAddressManagerBehaviourMode 

also these comments are not very useful in general
i can see those a lot in your code, not really happy about it :(

Reply:
I don't think when IPeerAddressManager is properly documented, that I need to rewrite all the comments? That just seems a bit superfluous.